Library/Homebrew/cmd/fetch.rb

```diff
@@ -32,7 +32,7 @@ def fetch
   def fetch_bottle? f
     return true if ARGV.force_bottle? && f.bottle
     return false unless f.bottle && f.pour_bottle?
-    if f.file_modified?
+    if f.file_modified? && !ARGV.build_from_source?
```

Comment:
I think it makes more sense to reorder this conditional with the one below it. Does that reduce the need to modify test-bot?